app/controllers/api/v1/votes_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Api::V1::VotesController < ApiController
+  def create
+    review_id = params[:review_id]
+    user_id = params[:current_user_id]
+    vote_value_string = params[:vote_value]
+    if vote_value_string == "upvote"
+      vote_value = 1
+    else
+      vote_value = -1
+    end
+
+    current_vote = Vote.where(user_id: user_id, review_id: review_id)[0]
+    if current_vote == nil
+      vote = Vote.new({review_id: review_id, user_id: user_id, value: vote_value})
+      if vote.save
+        render json: { vote: vote }
+      else
+        render json: { error: vote.errors.full_messages }, status: :unprocessable_entity
+      end
+    else
+      current_vote.value = vote_value
+      current_vote.save
+    end
```

Comment:
have you tested this `else` logic out? Generally, `.where` doesnt return a single record, but an array with one or more (even if there is only one vote that has those properties, it will be IN an array). So I would replace line 12's `.where` statement with `.find` instead, that way you will always get either one singular object or nil